pages/signin/index.tsx

```diff
@@ -0,0 +1,204 @@
+import React, { useCallback, useMemo, useState } from "react";
+import Head from "next/head";
+import {
+  Box,
+  Container,
+  TextField,
+  Typography,
+  Button,
+} from "@material-ui/core";
+import { gql, useMutation } from "@apollo/client";
+import { useRouter } from "next/router";
+
+const SEND_SIGN_TOKEN = gql`
+  mutation signin($email: String!, $username: String!) {
+    signin(email: $email, username: $username) {
+      message
+      token
+    }
+  }
+`;
+
+const IndexPage = () => {
+  const router = useRouter();
+  const [userEmail, setMail] = useState("");
+  const [userName, setName] = useState("");
+  const [errorMail, isErrorMail] = useState(false);
+  const [errorName, isErrorName] = useState(false);
+  const [messageErrorMail, setMessageErrorMail]= useState("");
+  const [messageErrorName, setMessageErrorName]= useState("");
+
+  const ableToSignIn = useMemo(() => {
+    return (
+      userEmail &&
+      userName &&
+      /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/.test(
+        userEmail
+      )
+    );
+  }, [userEmail, userName]);
+
+  const [addUser] = useMutation(SEND_SIGN_TOKEN);
+
+  const onClick = useCallback(() => {
+    if (!userEmail || !userName) return;
+
+    (async function () {
+      try {
+        await addUser({
+          variables: { email: userEmail, username: userName },
+        });
+        router.push("/auth");
+      } catch (e) {
+        if (e.message == "Already have a account") {
+          setMessageErrorName("")
+          isErrorName(false);
+          isErrorMail(true);
+          setMessageErrorMail(e.message)
+        } else {
+          setMessageErrorMail("")
+          isErrorMail(false);
+          isErrorName(true);
+          setMessageErrorName(e.message)
+        }
+      }
+    })();
+  }, [userEmail, userName]);
+
+  const onSubmit = useCallback((e) => {
```

Comment:
To trigger the sign-in on the form submit we need to call the onClick callback used on the button SIGN IN(line 196). Also considered change the name of the callback by something like `onSignIn`